bpfd/src/oci_utils/cosign.rs

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: (MIT OR Apache-2.0)
+// Copyright Authors of bpfd
+
+use std::str::FromStr;
+
+use anyhow::{anyhow, bail};
+use log::{debug, info, warn};
+use sigstore::{
+    cosign::{
+        verification_constraint::VerificationConstraintVec, verify_constraints, ClientBuilder,
+        CosignCapabilities,
+    },
+    errors::SigstoreError::RegistryPullManifestError,
+    registry::{Auth, ClientConfig, ClientProtocol, OciReference},
+    tuf::SigstoreRepository,
+};
+use tokio::task::spawn_blocking;
+
+pub struct CosignVerifier {
+    pub client: sigstore::cosign::Client,
+    pub allow_unsigned: bool,
+}
+
+impl CosignVerifier {
+    pub(crate) async fn new(allow_unsigned: bool) -> Result<Self, anyhow::Error> {
```

Comment:
No, because ultimately we'd just be wrapping Sigstore/Cosign's error types.
Typed errors are useful (to users) if they can pattern-match on them and do something useful about it.
In this case, all errors are fatal.

Reply:
Fair enough that argument is good enough for me, esp since I don't like setting up typed errors either 🤣 🚀